rotkehlchen/chain/ethereum/modules/balancer/decoder.py

```diff
@@ -0,0 +1,300 @@
+
+        return True
+
+    def _maybe_enrich_balancer_v1_events(
+        self,
+        token: 'EvmToken',
+        tx_log: EvmTxReceiptLog,  # pylint: disable=unused-argument
+        transaction: EvmTransaction,  # pylint: disable=unused-argument
+        event: HistoryBaseEntry,
+        action_items: list[ActionItem],  # pylint: disable=unused-argument
+        all_logs: list[EvmTxReceiptLog],
+    ) -> bool:
+        """
+        Enrich balancer v1 transfer to read pool events of:
+        - Depositing in the pool
+        - Withdrawing from the pool
+
+        In balancer V1 pools are managed using a DSProxy so the account doesn't interact
+        directly with the pools.
+        """
+        actions = self._decode_v1_pool_event(all_logs=all_logs)
+        if actions is None:
+            return False
+
+        for action in actions:
+            if (
+                action.extra_data is None or
+                'type' not in action.extra_data or
+                'ds_address' not in action.extra_data or
+                action.extra_data['ds_address'] != event.counterparty
+            ):
+                continue
+
+            if action.extra_data['type'] == BalancerV1EventTypes.JOIN:
+                if (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_type = HistoryEventType.STAKING
+                    event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                    event.counterparty = CPT_BALANCER_V1
+                    event.notes = f'Receive {event.balance.amount} {token.symbol} at {event.location_label} from Balancer V1 pool'  # noqa: E501
+                    return True
+                if (
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_type = HistoryEventType.STAKING
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                    event.counterparty = CPT_BALANCER_V1
+                    event.notes = f'Deposit {event.balance.amount} {token.symbol} in Balancer V1 pool from {event.location_label}'  # noqa: E501
+                    return True
+            elif action.extra_data['type'] == BalancerV1EventTypes.EXIT:
+                if (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_type = HistoryEventType.STAKING
+                    event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                    event.counterparty = CPT_BALANCER_V1
+                    event.notes = f'Receive {event.balance.amount} {token.symbol} in {event.location_label} after removing liquidity from a Balancer V1 pool'  # noqa: E501
```

Comment:
You seem to use different prepositions across different notes. Please be consistent.

```suggestion
                    event.notes = f'Receive {event.balance.amount} {token.symbol} at {event.location_label} after removing liquidity from a Balancer v1 pool'  # noqa: E501
```